### ISSUE TITLE
Unicode in R regular expressions

### ISSUE THREAD

**Issue by misterarias** [jupyter/docker-stacks#311]
## Summary

R is not installed with Unicode support for PCRE, and some regular expressions are not possible.
A notable use case is the `removeWords` function from the `tm` library.

![image](https://cloud.githubusercontent.com/assets/11632115/21105701/7ff655b4-c08b-11e6-812c-0ea3b8f10c3a.png)

## Scenario

While using the 'all-spark-notebook' container, if you type:

```{r}
string <- 'à]'
regex <- "(*UCP)\\w\\p{Pe}"
print(paste0("Matches: ", head(regmatches(string, regexpr(regex, string, perl = T)))))
```

I get the following error: 

```
Warning message in regexpr(regex, string, perl = T):
“PCRE pattern compilation error
	'this version of PCRE is not compiled with Unicode property support'
	at '(*UCP)\w\p{Pe}'”
```

![image](https://cloud.githubusercontent.com/assets/11632115/21105539/e3976ba4-c08a-11e6-8024-f49b4de0587b.png)

### Possible cause

Since PCRE dependency is installed by `conda install` and its hard-coded to PCRE1 8.39, I found no workaround to this other than *avoid using `conda install -c r r-base`* and install R using the R repository for Debian, and then manually adding the R kernel to Jupyter.

I'm pretty sure that there must be a better way than hacking my way out of the issue, any thoughts?

## Possible fix

My temporary workaround is here, will polish it moving forward: https://github.com/misterarias/docker-stacks/commit/bff59952aa1f8023cbb67d8288c6a98787881213 and 
https://github.com/misterarias/docker-stacks/commit/77adcf5293caafa8f469c177a468b4cc3c33c0cd

![image](https://cloud.githubusercontent.com/assets/11632115/21105561/f63e0c04-c08a-11e6-9ec8-b1996dbe7fa3.png)

**Comment by parente** [comment]
> I'm pretty sure that there must be a better way than hacking my way out of the issue, any thoughts?

When the docker-stacks project started, the thinking was that R package management with conda would be a nice parallel to Python. A year plus later, we might want to revisit.

I'm not an R expert by any stretch of the imagination. An evaluation of the best way forward from someone who knows the R packaging ecosystem well would be greatly appreciated.

**Comment by jakirkham** [comment]
We are working on adding R to conda-forge. Maybe some discussion about how to improve packaging of the R stack would be productive.

**Comment by jakirkham** [comment]
Also CC'ing @mingwandroid in case he has any input on this.

**Comment by parente** [comment]
Five months later and I see there are quite a few R packages on conda-forge now including r-base and PCRE. It might be worth a PR to see 1) how many packages are missing from conda-forge that the r-notebook stack is using from the anaconda r channel 2) if the PCRE unicode problem still exists.

**Comment by jakirkham** [comment]
I think issue ( https://github.com/conda-forge/staged-recipes/issues/2009 ) may be of interest here. This is being used to track the migration of R packages to conda-forge. Also we recently added a new reviewer to staged-recipes whose interest is integrating R into conda-forge. Hopefully this will streamline R migration into conda-forge. Looking at the r-notebook it looks like we have some of the packages integrated, but still have a few to go. Of those that remain outstanding, some of them are level 7 (topologically) so it may take a bit to get to them.

As to the issue at hand, it looks like `pcre` is used by `r-base`. Both of these are in conda-forge. If there is some change needed to the builds of either of these, feel free to submit a PR to the relevant feedstock and give me a ping so I'm aware. Then we can go from there.

**Comment by parente** [comment]
@jakirkham Thanks for the staged-recipes link. I'll definitely keep an eye on that.

**Comment by mingwandroid** [comment]
The actual bug here has been fixed since the 6th of December when I released the `pcre 8.39-1` package to conda defaults. This is exactly one day after this issue was opened. I fixed it because `glib` needed `pcre` to be built with Unicode Properties, so this one day difference was purely coincidental.

If this issue had been filed at https://github.com/ContinuumIO/anaconda-issues/issues instead of here (or if I had been cc'ed while not on vacation) then I'd probably have noticed it before today. Ah well.

**Comment by parente** [comment]
Not a problem @mingwandroid and thanks for reporting back in. I've been remiss in triaging open issues here.

I'm sure this has been fixed due to the numerous rebuilds of the images since December. @misterarias feel free to reopen if you're still seeing the issue.
